package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -455,19 +463,30 @@ def guess_donors(self, select='all', max_charge=-0.5):
             hydrogens_sel = self.hydrogens_sel
         hydrogens_ag = self.u.select_atoms(hydrogens_sel)
 
-        ag = hydrogens_ag.residues.atoms.select_atoms(
-            "({donors_sel}) and around {d_h_cutoff} {hydrogens_sel}".format(
-                donors_sel=select,
-                d_h_cutoff=self.d_h_cutoff,
-                hydrogens_sel=hydrogens_sel
+        if hasattr(hydrogens_ag[0],"bonded_atoms") and hydrogens_ag[0].bonded_atoms:
+            donors_ag = self.u.atoms[[x.bonded_atoms[0].ix for x in hydrogens_ag]]
```

Comment:
Might be worth reusing this function here: https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/analysis/hydrogenbonds/hbond_autocorrel.py#L223

Reply:
Thanks for the heads up!